### PR TITLE
fix #51 - Improve Bridge-gui devops

### LINKED ISSUE #51: Improve Bridge-gui devops
#### Really inconvenient to render changes

In order to see the changes you've made to the code in the browser, you have to `npm run build` and restart your `npm run start-dev` server. 

#### `clean-webpack-plugin` isn't working. 

This causes tons of minified code to pile up in the output directory. This is undesirable because it makes searching within the codebase harder and if using an IDE can fill up ram with a bunch of garbage.

`npm run build` outputs this message:
  ```
clean-webpack-plugin: /Users/bwhite/Projects/storj/bridge-gui/static/dist is outside of the project root.  Skipping...
  ```

### PR DESCRIPTION
**Pull request by bryanchriswhite**
#### `CleanWebpack` plugin not doing it's job:

Used to get "<path> is outside of project root. Skipping..." message

#### `webpack-isomorphic-tools` aren't really necessary

We're only doing partial server-side rendering of the index document. For this `webpack-isomorphic-tools` is unnecessary and adds additional configuration complexity.

#### `webpack-dev-middleware` and `webpack-hot-middleware` add unnecessary complexity

`webpack-dev-server` sits on top of express and can be used instead of `webpack-dev-middleware` and `webpack-hot-middleware`, as those middleware add additional configuration complexity as well as additional dependencies.

#### Docker tweaks

* comment out `RUN npm build` step in docker file as it's currently not needed (we may want to change this - needs discussion)
* fix possible es6/es5 issue in bin/server.js

### CODE REVIEW

**Comment by bryanchriswhite** [comment]
Despite the revert commit, github didn't re-open #53 so this is a dup... also, apparently this PR is only showing commits since the merge, again, despite the revert.

@phutchins 7eb3ec1135d4b27e7730b72ed5e53afdbeeb8736 has the Dockerfile changes i was telling you about which bumps the node version to 4.x. I'm not sure why it was at 0.12 before and/or how that was working.

Given the changes i've made to bin/server.js, webpack is building when bin/server.js executes in both dev and prod. This means that it's not necessary to run `npm run build` before `npm run start-prod`. This also means though that any time the server is restarted, webpack and all the deps are required, and the dist is rebuilt. This may or may not be a big deal given the nature of the system; personally, I think it doesn't really matter either way. I'd like to hear some feedback on that.

**Comment by bryanchriswhite** [comment]
so it looks like the revert that @super3 made created it's own branch/PR that didn't get merged... so since these commits are on the head of the commits that were prematurely merged there is no need to revert the revert.

For code review purposes though that means that ppl will have to reverence this PR as well as #53.

Further, we should close #54 and prune that branch to prevent any further error propagation.

All that said, @bookchin, @phutchins, @MrFancyMonocle, I think this PR is ready to go! I haven't added tests yet because the docker work took up the time I expected to spend on karma/phantom. I don't need tests to resume #29 so I would finish that so users who are affected by the password mis-hash issue can reset via the gui, then come back to tests quickly. @bookchin you think that makes sense?

**Comment by bookchin** [comment]
Sounds good to me. Let's merge after standup on Monday.

File: Dockerfile [hunk 1]
```diff
@@ -7,10 +7,12 @@ WORKDIR /opt/bridge-gui
 # Copy over the app and install
 COPY . /opt/bridge-gui/
 RUN /opt/bridge-gui/scripts/install_deps_debian.sh
-# Need to clean node_modules dir here or exclude it
-RUN npm install
-# Build the app
-RUN npm run build
+
```
Comment (bryanchriswhite) [inline Dockerfile]: @phutchins as per our discussion lets add a `git reset --hard && git clean -xf` here as well

File: bin/server.js [hunk 2]
```diff
@@ -1,31 +1,45 @@
...
+  global.__CLIENT__ = false;
+  global.__SERVER__ = true;
+  global.__DISABLE_SSR__ = true;  // <----- DISABLES SERVER SIDE RENDERING FOR ERROR DEBUGGING
+  global.__DEVELOPMENT__ = process.env.NODE_ENV !== 'production';
+
+  if (__DEVELOPMENT__) {
+    console.log('RUNNING IN DEVELOPMENT');
+    require('../webpack/webpack-dev-server');
+  } else {
+    var webpack = require('webpack');
```
Comment (bryanchriswhite) [inline bin/server.js]: don't build here

**Comment by phutchins** [comment]
Reviewed this and other included PR. All looks good. Plan on merging this shortly.

**Pull request merged**